eth/db/account.py

```diff
@@ -422,9 +441,15 @@ def persist(self) -> None:
                     f"is missing for hash 0x{new_root.hex()}."
                 )
 
+        # generate witness (copy) before clearing the underlying data
+        witness_index = self._get_witness_index()
+
         # reset local storage trackers
         self._account_stores = {}
         self._dirty_accounts = set()
+        self._accessed_accounts = set()
+        self._accessed_bytecodes = set()
+        self._account_cache.clear()
```

Comment:
Why was this call to `clear()` added?

Reply:
If we don't clear, then subsequent accesses that look up an account may not get added to the access list. (We skip adding the account to the access list if it's in cache) I'll add a comment explaining.